### PR TITLE
Add page replacement algorithms

### LINKED ISSUE #1554: [Idea]: <Your idea title>Add page replacement algorithms
### Idea Title

Add page replacement algorithms

### Idea Description

Page replacement algorithms are essential in operating systems for managing memory when physical RAM cannot accommodate all active pages. They determine which pages to evict when new ones need to be loaded. Common algorithms include Least Recently Used (LRU), which replaces the least recently accessed page; First-In, First-Out (FIFO), which evicts the oldest page; and Optimal Page Replacement, which removes the page that won't be used for the longest time. Each algorithm has its pros and cons, with LRU often being more efficient than FIFO but more complex to implement. Understanding these algorithms is crucial for optimizing memory usage and reducing page faults.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [ ] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by Ananya-vastare**
### Description
Least Recently Used (LRU) and First-In, First-Out (FIFO) page replacement algorithms by enhancing the system's memory management capabilities. With LRU prioritizing pages based on recent usage and FIFO removing the oldest page first, this change provides users with a comparative analysis of these widely used algorithms to optimize page replacement performance. This feature aims to improve system efficiency when handling memory references, making it a valuable addition to the project. No additional dependencies are required for this implementation.

Fixes #1554 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Ananya-vastare** [comment]
@ajay-dhangar please let me know if I need to make any changes

**Comment by Ananya-vastare** [comment]
@ajay-dhangar any updates please

File: docs/Operating System/pagereplacementaglorithms.ms [hunk 1]
```diff
@@ -0,0 +1,147 @@
+id: FIFO-LRU title: "FIFO and LRU Page Replacement Algorithms" sidebar_label: "FIFO and LRU" sidebar_position: 1 description: "A comprehensive guide to understanding and implementing FIFO and LRU Page Replacement Algorithms in memory management." tags: [fifo, lru, memory management, page replacement, os]
+FIFO and LRU Page Replacement Algorithms
+Page replacement algorithms are essential in operating systems for managing memory efficiently when there is limited space. FIFO (First-In, First-Out) and LRU (Least Recently Used) are two fundamental page replacement techniques.
```
Comment (ajay-dhangar) [inline docs/Operating System/pagereplacementaglorithms.ms]: not a good format

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
First you need save your file with `.md` extension and follow our format

**Comment by Ananya-vastare** [comment]
Okay will do that by the end of the day @ajay-dhangar

**Comment by Ananya-vastare** [comment]
@ajay-dhangar please know if I need to make any changes

**Comment by Ananya-vastare** [comment]
@ajay-dhangar please let me know if I need to make any changes

File: Operating System/pagereplacementalgorithm.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
+# Page Replacement Algorithm: Least Recently Used (LRU)
```
Comment (ajay-dhangar) [inline Operating System/pagereplacementalgorithm.md]: Please follow the syntax and format for our docs. If you have any questions, please don't hesitate to ask.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Update your code based on suggestions

**Pull request closed without merging**